NavigationReactNative/src/ios/NVSegmentedTabView.m

```diff
@@ -16,16 +19,34 @@ - (id)init
     return self;
 }
 
-- (void)setTitles:(NSArray<NSString *> *)titles
+- (void)setTabs:(NSArray<NSDictionary *> *)tabs
 {
+    segmentTestIDs = [[NSMutableDictionary alloc] init];
     NSInteger selectedIndex = self.selectedSegmentIndex;
     [self removeAllSegments];
-    for (NSString *title in titles) {
-        [self insertSegmentWithTitle:title atIndex:self.numberOfSegments animated:NO];
+    for (NSDictionary *tab in tabs) {
+        NSUInteger index = self.numberOfSegments;
+        [self insertSegmentWithTitle:tab[@"title"] atIndex:index animated:NO];
+        UIView *segment = [self subviews][index];
+        if (segment) {
+            segmentTestIDs[[NSValue valueWithNonretainedObject:segment]] = tab[@"testID"];
+        }
```

Comment:
I’ve got a way to set the `accessibilityIdentifier` inside the loop of `setTabs`. The trick is to use the `accessibilityElements`. These point at the `UISegments` so you can then get at the labels and set the `accessibilityIdentifiers`. I know you had problems with these identifiers getting removed which was why you had to set them later. But, surprisingly, they aren’t removed when using the `accessibilityElements`!

```obj-c
UIView *segment = ((UIView *) self.accessibilityElements[index]);
segment.subviews[0].accessibilityIdentifier = tabs[index][@"testID"];
```

Reply:
Hmm I wasn't able to get this working, could be an issue with my Xcode version? I'm on 11.7/iOS 13.7 for most of this.

Reply:
Looks good. I think I only tried in iOS 14. What goes wrong?

Reply:
Aargh, just tried it again and it's not working. Leave it with me

Reply:
Ah okay, I just tried it on Xcode 12.2 / iOS 14.2 simulator and same thing. Let me know if you have any luck!

Reply:
Ok, it works inside the `layoutSubviews`. You just need to store the `tabs` so they're available to the `layoutSubviews`. It's better than looping through the subviews because the `accessibilityElements` are in the same order as the tabs - so can access by index
```obj-c
- (void)layoutSubviews{
    [super layoutSubviews];
    for(int i = 0; i < self.accessibilityElements.count; i++) {
        UIView *segment = ((UIView *) self.accessibilityElements[i]);
        segment.subviews[0].accessibilityIdentifier = tabs[i][@"testID"];
    }
}
```